mqtt/mqtt-bridge/src/persist/waking_state/rocksdb.rs

```diff
@@ -0,0 +1,146 @@
+use std::{
+    collections::{hash_set::HashSet, HashMap, VecDeque},
+    task::Waker,
+    vec::IntoIter,
+};
+
+use bincode::{self};
+use mqtt3::proto::Publication;
+use rocksdb::{ColumnFamily, IteratorMode, Options, DB};
+use uuid::Uuid;
+
+use crate::persist::{waking_state::StreamWakeableState, Key, PersistError};
+
+/// When elements are retrieved they are added to the in flight collection, but kept in the original db store.
+/// Only when elements are removed from the in-flight collection they will be removed from the store.
+pub struct WakingRocksDBStore {
+    db: RocksDbWrapper,
+    in_flight: HashMap<Key, Publication>,
+    waker: Option<Waker>,
+}
+
+impl WakingRocksDBStore {
+    pub fn new(db: DB) -> Result<Self, PersistError> {
+        let db = RocksDbWrapper::new(db)?;
+
+        Ok(Self {
+            db,
+            in_flight: HashMap::new(),
+            waker: None,
+        })
+    }
+}
+
+impl StreamWakeableState for WakingRocksDBStore {
+    fn insert(&mut self, key: Key, value: Publication) -> Result<(), PersistError> {
+        self.db.insert(&key, &value)?;
+        if let Some(waker) = self.waker.take() {
+            waker.wake();
+        }
+
+        Ok(())
+    }
+
+    /// Get count elements of store, exluding those that are already in in-flight
+    fn batch(&mut self, count: usize) -> Result<VecDeque<(Key, Publication)>, PersistError> {
+        let iter = self
+            .db
+            .iter_except(count, &self.in_flight.keys().collect())?;
```

Comment:
I would just pass HashMap and use `contains_key`, to save some cpu cycles on creating a separate HashSet.
```suggestion
            .iter_except(count, &self.in_flight)?;
```